File: benchmarks/bench_bitflip.c
#include <stdint.h>
static int32_t flipbits(int32_t x) { return ~x; }
static int popcount(int32_t x) {
    int c=0;
    for (int b=0; b<32; b++) { if ((x & (1<<b)) == (1<<b)) c++; }
    return c;
}
int main(void) {
    int32_t arr[8]={5,3,8,1,9,2,7,4};
    int ok=1;
    for (int r=0; r<1000000; r++)
        for (int i=0; i<8; i++)
            if (popcount(arr[i])+popcount(flipbits(arr[i])) != 32) ok=0;
    return ok ? 0 : 1;
}

File: benchmarks/bench_bitflip.zig
fn flipbits(x: i32) i32 { return ~x; }
fn popcount(x: i32) i32 {
    var c: i32 = 0;
    var b: i32 = 0;
    while (b < 32) : (b += 1) {
        const m: i32 = @as(i32,1) << @intCast(b);
        if ((x & m) == m) c += 1;
    }
    return c;
}
pub fn main() u8 {
    const arr = [8]i32{5,3,8,1,9,2,7,4};
    var ok: bool = true;
    var r: usize = 0;
    while (r < 1000000) : (r += 1) {
        for (arr) |v| if (popcount(v) + popcount(flipbits(v)) != 32) { ok = false; };
    }
    return if (ok) 0 else 1;
}
